src/main/scala/mesosphere/marathon/tasks/PortsMatcher.scala

```diff
@@ -40,27 +40,42 @@ class PortsMatcher(
 
   lazy val portsMatch: Option[PortsMatch] = portsWithRoles.map(PortsMatch(_))
 
+  //scalastyle:off cyclomatic.complexity
   private[this] def portsWithRoles: Option[Seq[Option[PortWithRole]]] = {
-    val portMappings: Option[Seq[Container.Docker.PortMapping]] =
-      for {
-        c <- runSpec.container
-        pms <- c.portMappings if pms.nonEmpty
-      } yield pms
-
-    (runSpec.portNumbers, portMappings) match {
-      case (Nil, None) => // optimization for empty special case
-        Some(Seq.empty)
-
-      case (ports, Some(mappings)) =>
-        // We use the mappings from the containers if they are available and ignore any other port specification.
-        // We cannot warn about this because we autofill the ports field.
-        mappedPortRanges(mappings)
-
-      case (ports, None) if runSpec.requirePorts =>
-        findPortsInOffer(ports, failLog = true)
-
-      case (ports, None) =>
-        randomPorts(ports.size)
+    runSpec match {
+      case podDefinition: PodDefinition =>
+        val requiredPorts = for {
+          container <- podDefinition.containers
+          endpoint <- container.endpoints
+        } yield endpoint.hostPort
+
+        if (podDefinition.requirePorts) {
```

Comment:
I don't think `requirePorts` is meaningful for pods. We should always be using `mappedPortRanges`